src/app/shared/form/copy-input.directive.ts

```diff
@@ -0,0 +1,36 @@
+import { Directive, ElementRef, Input, OnInit, HostListener } from '@angular/core';
+
+@Directive({
+  selector: '[publishCopyInput]'
+})
+export class CopyInputDirective implements OnInit {
+
+  @Input() publishCopyInput: HTMLInputElement;
+
+  private startText = 'Copy';
+  private copiedText = 'Copied';
+
+  constructor(private el: ElementRef) {}
+
+  @HostListener('mouseover') onMouseOver(): void {
+    this.el.nativeElement.innerHTML = this.startText;
+  }
+
+  @HostListener('click') onClick(): void {
+    if (this.publishCopyInput && this.publishCopyInput.select) {
```

Comment:
That does sound familiar.  I didn't notice any compilation errors on this, though.  Maybe because I'm passing it in via this directive's `@Input`... typescript doesn't have any choice but to trust me on what type it is?